New qml_lint picks up Qt 5 version, fails the build

```
# kdesrc-build running: 'ninja' '-j' '6'
# from directory: /home/ratijas/kde/build6/gammaray
[1/24] Run qmlint on the specified files and create the associated timestamp semaphore
FAILED: examples/quick-batching/Slider.qml.qmllint /home/ratijas/kde/build6/gammaray/examples/quick-batching/Slider.qml.qmllint 
cd /home/ratijas/kde/build6/gammaray/examples/quick-batching && /usr/bin/qmllint /home/ratijas/kde/src6/gammaray/examples/quick-batching/Slider.qml && /usr/bin/cmake -E touch /home/ratijas/kde/build6/gammaray/examples/quick-batching/Slider.qml.qmllint
/home/ratijas/kde/src6/gammaray/examples/quick-batching/Slider.qml:16 : Expected token `;'
[2/24] Run qmlint on the specified files and create the associated timestamp semaphore
[3/24] Linking CXX shared module lib/gammaray/3.0/qt6_6-x86_64/gammaray_kjobtracker_plugin.so
[4/24] Building CXX object ui/CMakeFiles/gammaray_ui.dir/aboutdata.cpp.o
[5/24] Building CXX object translations/CMakeFiles/ljsonupdate.dir/ljsonupdate.cpp.o
[6/24] Building CXX object launcher/cli/CMakeFiles/gammaray.dir/main.cpp.o
ninja: build stopped: subcommand failed.
```

Clearly, /usr/bin/qmllint is coming from qt5-declarative package.

Besides, what the heck is this linter suggesting? It is really suggesting no whitespace between the => arrow function?

```js
            onClicked: (mouse)=> {
                updateCurrentIndex(mouse.x, mouse.y)
            }
```

--- a/cmake/FindQmlLint.cmake
+++ b/cmake/FindQmlLint.cmake
@@ -8,49 +8,65 @@
 # Contact KDAB at <info@kdab.com> for commercial licensing options.
 #
 
-find_program(QMLLINT_EXECUTABLE qmllint)
-if(QMLLINT_EXECUTABLE)
-    if(NOT QMLLINT_IS_WORKING)
-        # Try to fix common problems on Debian-based distros -- they provide /usr/bin/qmllint,
-        # which is a symlink to /usr/lib/x86_64-linux-gnu/qt4/bin/qmllint (or the Qt5 version of it).
-        # The actual executable is part of different package, so might not even be installed =>
-        # double-check whether qmllint is working by executing it
-        execute_process(
-            COMMAND ${QMLLINT_EXECUTABLE} --version
-            RESULT_VARIABLE _qmllint_result
-            OUTPUT_QUIET ERROR_QUIET
-        )
-        if(_qmllint_result EQUAL 0)
-            set(QMLLINT_IS_WORKING
-                TRUE
-                CACHE BOOL "Whether the found qmllint executable is actually usable" FORCE
+set(QmlLint_EXECUTABLE)
+set(QmlLint_FOUND FALSE)
+
+# First check for a target (Qt6)
+if(TARGET Qt${QT_VERSION_MAJOR}::qmllint)
+    set(QmlLint_FOUND TRUE)
+    set(QmlLint_EXECUTABLE Qt${QT_VERSION_MAJOR}::qmllint)
+else()
+    # See if it's on the PATH
+    find_program(QmlLint_EXECUTABLE qmllint)
+    if(QmlLint_EXECUTABLE)
+        if(NOT QMLLINT_IS_WORKING)
+            # Try to fix common problems on Debian-based distros -- they provide /usr/bin/qmllint,
+            # which is a symlink to /usr/lib/x86_64-linux-gnu/qt4/bin/qmllint (or the Qt5 version of it).
+            # The actual executable is part of different package, so might not even be installed =>
+            # double-check whether qmllint is working by executing it
+            execute_process(
+                COMMAND ${QmlLint_EXECUTABLE} --version
+                RESULT_VARIABLE _qmllint_result
+                OUTPUT_QUIET ERROR_QUIET
             )
+            if(_qmllint_result EQUAL 0)
+                set(QMLLINT_IS_WORKING
+                    TRUE
+                    CACHE BOOL "Whether the found qmllint executable is actually usable" FORCE
+                )
+            endif()
         endif()
-    endif()
-    if(QMLLINT_IS_WORKING)
-        set(QmlLint_FOUND TRUE)
+        if(QMLLINT_IS_WORKING)
+            set(QmlLint_FOUND TRUE)
+        endif()
     endif()
 endif()
 
 # validate a list of qml files
 function(qml_lint)
-    if(NOT QMLLINT_EXECUTABLE OR NOT QmlLint_FOUND)
+    if(NOT QmlLint_EXECUTABLE OR NOT QmlLint_FOUND)
         return()
+    endif()
+
+    if(NOT TARGET qmllint_all)
+        add_custom_target(qmllint_all ALL COMMENT "Scan all .qml files with qmllint")
     endif()
 
     foreach(_file ${ARGN})
         get_filename_component(_file_abs ${_file} ABSOLUTE)
         add_custom_command(
-            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${_file}.qmllint
-            COMMAND ${QMLLINT_EXECUTABLE} ${_file_abs}
-            COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/${_file}.qmllint
+            OUTPUT ${_file}.lint
+            COMMAND ${QmlLint_EXECUTABLE} ${_file_abs}
+            COMMAND ${CMAKE_COMMAND} -E touch ${_file}.lint
             MAIN_DEPENDENCY ${_file_abs}
-            COMMENT "Run qmlint on the specified files and create the associated timestamp semaphore"
+            WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+            COMMENT "Run qmllint on ${_file}"
         )
         add_custom_target(
-            ${_file}_qmllint ALL
-            DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${_file}.qmllint
-            COMMENT "Target to run qmllint on the specified file"
+            ${_file}_qmllint
+            DEPENDS ${_file}.lint
+            COMMENT "Ensure qmllint is run on ${_file}"
         )
+        add_dependencies(qmllint_all ${_file}_qmllint)
     endforeach()
 endfunction()

